speculation-rules.bs

```diff
@@ -253,15 +268,22 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
     1. &#x231B; [=list/For each=] |rule| of |ruleSet|'s [=speculation rule set/prerender rules=]:
       1. &#x231B; [=list/For each=] |url| of |rule|'s [=speculation rule/URLs=]:
         1. &#x231B; [=list/Append=] a [=prerender candidate=] with [=prerender candidate/URL=] |url| and [=prerender candidate/referrer policy=] "`strict-origin-when-cross-origin`" to |prerenderCandidates|.
+  1. &#x231B; [=list/For each=] |startedPrefetch| of |document|'s [=Document/started prefetches=]:
+    1. &#x231B; If |startedPrefetch| is not [=started prefetch/continued by=] any element of |prefetchCandidates|, then:
+      1. &#x231B; [=prefetch controller/Abort=] |startedPrefetch|'s [=started prefetch/controller=].
+      1. &#x231B; [=list/Remove=] |startedPrefetch| from |document|'s [=Document/started prefetches=].
   1. End the [=synchronous section=], continuing the remaining steps [=in parallel=].
   1. [=list/For each=] |prefetchCandidate| of |prefetchCandidates|:
-    1. The user agent may [=prefetch=] given |document|, |prefetchCandidate|'s [=prefetch candidate/URL=], |prefetchCandidate|'s [=prefetch candidate/referrer policy=] and |prefetchCandidate|'s [=prefetch candidate/anonymization policy=].
+    1. The user agent may run the following steps:
+      1. Let |prefetchController| be a new [=prefetch controller=] whose [=prefetch controller/document=] is |document|.
+      1. [=list/Append=] a new [=started prefetch=] whose [=started prefetch/candidate=] is |prefetchCandidate| and [=started prefetch/controller=] is |prefetchController| to |document|'s [=Document/started prefetches=].
+      1. [=Prefetch=] given |document|, |prefetchCandidate|'s [=prefetch candidate/URL=], |prefetchCandidate|'s [=prefetch candidate/referrer policy=], |prefetchCandidate|'s [=prefetch candidate/anonymization policy=], and |prefetchController|.
```

Comment:
It seems like it'd be cleaner if "prefetch" automatically updated the document's "started prefetches"... I guess that might make the document factoring and layering less good, though? Not sure...

Reply:
It now does so (well, each prefetch algorithm does so -- even though it's duplicated, it means that it would be possible to directly invoke either of them in the future).